src/utils/ouia.rs

```diff
@@ -0,0 +1,26 @@
+use std::sync::Mutex;
+
+pub struct Ouia(&'static str);
+
+impl Ouia {
+    pub const fn new(component_name: &'static str) -> Self {
+        Self(component_name)
+    }
+
+    pub fn generated_id(&self) -> String {
+        let count = counter();
+        format!("OUIA-Generated-{}-{count}", self.0)
+    }
+
+    pub fn component_type(&self) -> String {
+        format!("PF5/{}", self.0)
+    }
+}
+
+fn counter() -> usize {
+    static COUNT: Mutex<usize> = Mutex::new(0);
```

Comment:
Not sure that works, but why not use `Arc<AtomicUsize>>` to make this lock free? Or something like that.

Reply:
Doesn't even need the `Arc` since we aren't sharing the reference itself. The code simplifies to

```rust
fn counter() -> usize {
    static COUNT: AtomicUsize = AtomicUsize::new(0);
    COUNT.fetch_add(1, Ordering::Relaxed)
}
```

Reply:
Cool, I wasn't sure that would work. Yay :grinning: